src/interface/web/app/search/page.tsx

```diff
@@ -247,14 +285,38 @@ export default function Search() {
                         <div className="md:w-3/4 sm:w-full mx-auto pt-6 md:pt-8">
                             <div className="p-4 md:w-3/4 sm:w-full mx-auto">
                                 <div className="flex justify-between items-center border-2 border-muted p-1 gap-1 rounded-lg">
-                                    <Input
-                                        autoFocus={true}
-                                        className="border-none pl-4"
-                                        onChange={(e) => setSearchQuery(e.currentTarget.value)}
-                                        onKeyDown={(e) => e.key === "Enter" && search()}
-                                        type="search"
-                                        placeholder="Search Documents"
-                                    />
+                                    <div className="relative flex-1">
+                                        <Input
+                                            autoFocus={true}
+                                            className="border-none pl-4"
+                                            onChange={(e) => handleSearchInputChange(e.currentTarget.value)}
+                                            onKeyDown={(e) => {
+                                                if (e.key === "Enter") {
+                                                    if (showSuggestions && fileSuggestions.length > 0) {
+                                                        applySuggestion(fileSuggestions[0]);
+                                                    } else {
+                                                        search();
+                                                    }
+                                                }
+                                            }}
+                                            type="search"
+                                            placeholder="Search Documents (type 'file:' for file suggestions)"
+                                            value={searchQuery}
+                                        />
+                                        {showSuggestions && fileSuggestions.length > 0 && (
+                                            <div className="absolute z-10 w-full mt-1 bg-background border rounded-md shadow-lg">
```

Comment:
The options dropdown expands to the right edge of the screen, which feels a bit too wide in my opinion.

Maybe we can use the same `Command` and `Popover` dropdown UI components here? You can check how the slash commands uses them in `chatInputArea.tsx`.  
https://github.com/khoj-ai/khoj/blob/8fe08eecce42465fbb4f3164072dbf17b645f34c/src/interface/web/app/components/chatInputArea/chatInputArea.tsx#L497-L501

If that is too annoying to use, it may still be a good idea to restrict width of the file filter dropdown to not expand to the right edge of the screen.

Reply:
I wasn't able to reproduce the bug you're mentioning. In my view, the suggestions are only as wide as the input box. Let's sync up about it separately.